source/kraken/apply_disruption.cpp

```diff
@@ -716,8 +722,7 @@ struct add_impacts_visitor : public apply_impacts_visitor {
             mvj->push_unique_impact(impact);
 
             auto nb_rt_vj = mvj->get_vjs_at(rt_level).size();
-            std::string new_vj_uri = "vehicle_journey:" + mvj->uri + ":" + type::get_string_from_rt_level(rt_level)
-                                     + ":" + std::to_string(nb_rt_vj) + concatenate_impact_uris(*mvj);
+            auto new_vj_uri = make_new_vj_uri(mvj->uri, rt_level, nb_rt_vj, impact->disruption->uri);
```

Comment:
Not sure at all: Aren't you afraid of a case where multiple impacts of the same disruption are applied to the same VJ?
I think of something like below.

1. First impact:
uri = vj:mvj_uri:adapted:`1`:disruption_uri

2. Second impact (if vj cleanup is done after):
uri = vj:mvj_uri:adapted:`2`:disruption_uri
cleanup vj:mvj_uri:adapted:`1`:disruption_uri

3. Third impact: 
uri = vj:mvj_uri:adapted:`2`:disruption_uri -> conflict

From the way VJ are created, it may be silently replacing the previous with the same uri, and it may work... Or not.
Should you at least log if there is an uri conflict?

Maybe I'm just being paranoïd (about RT in kraken, i am :exploding_head:)

Reply:
You're right. 

During the very first test, I was even more aggressive, I removed even disruption_uri from the new vj's URI. Kraken did crash.

Then I realized the very same scenario that you described had happened.  The reason why it works with only disruption_uri is in the same disruption, there **should not be more than one impact** that impacts the same object. 

Now your comment gave me a second thoughts. May be we should keep `concatenate_impact_uris(*mvj)` in new vj's uri.

Reply:
or for the sake speed, we can just use a uuid...

Reply:
As discussed, I just add some blabla here (really tracked in JIRA: https://navitia.atlassian.net/browse/NAV-1800): this UUID generation is preventing from requesting a given RT-VJ from its id on another kraken of the same coverage.